### PR TITLE
Remove where.exe timeout

### PR DESCRIPTION
**Pull request by pakrym**
to 10 seconds

https://github.com/aspnet/IISIntegration/issues/939

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h [hunk 1]
```diff
@@ -96,5 +96,7 @@ class HOSTFXR_UTILITY
 
     static
     std::wstring ExpandEnvironmentVariables(const std::wstring & str);
+
+    static const DWORD m_WhereExeProcessTimeout = 10 * 1000;
```
Comment (Tratcher) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: Name or comment that the units are seconds.

File: src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h [hunk 2]
```diff
@@ -96,5 +96,7 @@ class HOSTFXR_UTILITY
 
     static
     std::wstring ExpandEnvironmentVariables(const std::wstring & str);
+
+    static const DWORD m_WhereExeProcessTimeoutMS = 10 * 1000;
```
Comment (pan-wang) [inline src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: what's the motivation to increate the timeout to 10s? should we consider a environment variable for the value if we cannot determine the max timeout value

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: Motivation is that server should not be failing to start on slow machines.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: Calling where.exe was timing out on our CI (The main reason we saw test failures). We could make the timeout configurable.

Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: I don't think we should make it configurable. Maybe we should just increase it to a minute. 

The side effect of where.exe timing out is often app not starting.

Comment (pan-wang) [reply src/AspNetCoreModuleV2/CommonLib/hostfxr_utility.h]: I meant whether there would be any AV or crash.

**Comment by pakrym** [comment]
After talking to @mikeharder I removed `where.exe` timeout all together. Not getting a result of `where.exe` and continuing just leads to bugs and startup flakiness. IIS is free to kill worker process if it doesn't start in time.

**Comment by mikeharder** [comment]
Can we add a test for this behavior?  We could use a fake `where.exe` which never returns, and test what IIS does in this scenario.

**Comment by pakrym** [comment]
We don't have full IIS testing infrastructure yet.

**Comment by mikeharder** [comment]
@jkotalik: Fine with adding tests in separate PR, we can just create an issue for the test debt and prioritize it accordingly.  Might even want a separate project for these tests if they're very slow or different from other tests.

**Comment by mikeharder** [comment]
@pakrym: Is there a big difference between IIS and IIS Express with regards to startup timeout?  When I said "IIS" I meant "IIS and/or IIS Express".

Issue to add tests for startup timeouts: https://github.com/aspnet/IISIntegration/issues/972

**Comment by pakrym** [comment]
Yes, when we use IIS Express in tests we start `iisexpress.exe` process manually, full IIS has an orchestrator that coordinates worker processes and startup timeout is implemented in it.

It's not possible to test features related to the process management using IIS Express.

**Pull request merged**